gli/utils.py

```diff
@@ -392,3 +392,366 @@ def save_data(prefix, **kwargs):
+    
+    def save_metadata(self):
+        metadata = {"description" : "%s dataset" % self._name,
+         "Node" : self._node,
+         "Edge" : self._edge,
+         "Graph" : self._graph,
+         "citation" : self._citation,
+         "is_heterogeneous" : self._is_heterogeneous}
+        with open("metadata.json","w") as f:
+            json.dump(metadata,f)
+    
+    def _base_task(
+        self,
+        description="",
+        feature=[],
+        target="",
+        num_classes=0,
+        num_splits=1,
+        train_set=None,
+        val_set=None,
+        test_set=None,
+        num_samples=0,
+        train_ratio=0,
+        val_ratio=0,
+        test_ratio=0
+    ):
+        missing_features_messages_error = []
+        missing_features_messages_warning = []
+        self._description = description
+        self._feature = feature
+        self._target = target
+        self._num_classes = num_classes
+        self._num_splits = num_splits
+        self._num_samples = num_samples
+        self._train_ratio = train_ratio
+        self._val_ratio = val_ratio
+        self._test_ratio = test_ratio
+        if train_set:
+            if "file" not in train_set:
+                missing_features_messages_warning.append("%s missing 'file' in 'train_set'" % self._name)
+                train_set.update({"file" : "%s_task.npz" % self._name})
+            if "key" not in train_set:
+                missing_features_messages_warning.append("%s missing 'key' in 'train_set'" % self._name)
+                train_set.update({"key" : "train"})
+        self._train_set = train_set
+        if val_set:
+            if "file" not in val_set:
+                missing_features_messages_warning.append("%s missing 'file' in 'val_set'" % self._name)
+                val_set.update({"file" : "%s_task.npz" % self._name})
+            if "key" not in val_set:
+                missing_features_messages_warning.append("%s missing 'key' in 'val_set'" % self._name)
+                val_set.update({"key" : "val"})
+        self._val_set = val_set
+        if test_set:
+            if "file" not in test_set:
+                missing_features_messages_warning.append("%s missing 'file' in 'test_set'" % self._name)
+                test_set.update({"file" : "%s_task.npz" % self._name})
+            if "key" not in test_set:
+                missing_features_messages_warning.append("%s missing 'key' in 'test_set'" % self._name)
+                test_set.update({"key" : "val"})
+        self._test_set = test_set
+
+        if ((self._num_samples != 0) or (self._train_ratio != 0) or (self._val_ratio != 0) or (self._test_ratio!= 0)) and (self._train_set or self._val_set or self._test_set):
+            missing_features_messages_error.append("%s has conflict in splits in %s" % (self._name, self._type))
+
+        return missing_features_messages_error, missing_features_messages_warning
```

Comment:
Perhaps we can directly print the error at the end of the function and raise a
`ValueError` then.